File: calib_front/src/utils/datatypes.ts
export type Side = 'left' | 'right'

export type Point = {
    x: number;
    y: number;
}

export type Pair = {
    id: number;
    selected: boolean;
}

export type Existence = {
    exists: boolean;
    imgId: number;
}

export type Metadata = {
    id: number;
    side: Side;
    foundCorners: boolean;
    corners: Array<Point>;
}

File: calib_back/src/datatypes.py
from pydantic import BaseModel

class Pair(BaseModel):
    id: int

class Point(BaseModel):
    x: float
    y: float

class ImageMetadata(BaseModel):
    foundCorners: bool
    corners : list[Point]

class SelectedPairs(BaseModel):
    selectedIds: list[int]